dataprofiler/profilers/categorical_column_profile.py

```diff
@@ -22,7 +27,7 @@ class CategoricalColumn(BaseColumnProfiler):
     # Default value that determines if a given col is categorical or not.
     _CATEGORICAL_THRESHOLD_DEFAULT = 0.2
 
-    def __init__(self, name, options=None):
+    def __init__(self, name: Optional[str], options: CategoricalOptions = None) -> None:
```

Comment:
I'm not sure `name` is allowed to be optional. Reason for choosing Optional here?

Reply:
as above

Reply:
same reason as above in the `__add__` method